pkg/specgen/generate/kube/volume.go

```diff
@@ -169,7 +169,7 @@ func VolumeFromSecret(secretSource *v1.SecretVolumeSource, secretsManager *secre
 
 	err = yaml.Unmarshal(secretByte, secret)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("only secrets created via the kube yaml file are supported: %w", err)
```

Comment:
when you go back and add a test, maybe make this `if err := yaml.Unmarshal(secretByte, secret); err != nil { ...`  Not required.